x/ibc/20-transfer/types/packet.go

```diff
@@ -59,11 +59,15 @@ func (ftpd FungibleTokenPacketData) GetBytes() []byte {
 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(ftpd))
 }
 
-// AckDataTransfer is a no-op packet
+// FungibleTokenPacketAcknowledgement contains a boolean success flag and an optional error msg
+// error msg is empty string on success
 // See spec for onAcknowledgePacket: https://github.com/cosmos/ics/tree/master/spec/ics-020-fungible-token-transfer#packet-relay
-type AckDataTransfer struct{}
+type FungibleTokenPacketAcknowledgement struct {
+	Success bool   `json:"success" yaml:"success"`
+	Error   string `json:"error" yaml:"error"`
```

Comment:
q: why not an `error` type instead? then you can remove the success field and do:

```go
if ack.Error() != nil {
    return ack.Error()
}
// success
```

Reply:
Just went off the spec. Agreed this has the same behavior.

@cwgoes any danger in switching to the above?

Reply:
No, no danger, it just needs to be serialised correctly.